### PR TITLE
#10 add authorization

### PR DESCRIPTION
**Pull request by MattGomb**
Dear Reviewer,

welcome

In this PR we followed the instructions of [Blog app: add authorization rules](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_add_authorization.md):

- Install CanCanCan in your project.
- Add a role column to the users table. Remember to use a migration for this.
  A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this 
  authorization.
- For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only 
authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    # class of the resource.
+    #
+    # The third argument is an optional hash of conditions to further filter the
+    # objects.
+    # For example, here the user can only update published articles.
+    #
+    #   can :update, Article, published: true
+    #
+    # See the wiki for details:
+    # https://github.com/CanCanCommunity/cancancan/blob/develop/docs/define_check_abilities.md
```
Comment (tufoinnkuo10) [inline app/models/ability.rb]: - [x] kindly remove the comments to keep your code clean and professional. nice work

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -33,6 +34,22 @@ def create
     end
   end
 
+  def destroy
```
Comment (tufoinnkuo10) [inline app/controllers/posts_controller.rb:39]: - [x] great job for implementing the destroy actions in the PostsController and  CommentsController. please kindly Use the CanCanCan for Post authorization in Post Controller.
eg 
```ruby
 class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @team

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Git flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally

Required Changes ♻️
Check the comments under the review

tufoin

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (MattGomb) [reply app/controllers/posts_controller.rb:39]: @tufoinnkuo10 can you explain why, and can you point to which requirement asks for this?

File: app/models/ability.rb (hunk 1 above)
Comment (MattGomb) [reply app/models/ability.rb]: Done!

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (MattGomb) [reply app/controllers/posts_controller.rb:39]: @tufoinnkuo10 load_and_authorize_resource breaks the app this way, why would I use it...?

**Review by victorokoroji — APPROVED** [review]
Hi @MattGomb, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request is following the Git Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- The commit messages are meaningful ✅ 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**